### PR TITLE
ignoredResources should contain events.k8s.io events

### LINKED ISSUE #108318: controller-manager gc sync event failed
### What happened?

15:30:38.355384    2136 graph_builder.go:279] garbage controller monitor not yet synced: events.k8s.io/v1, Resource=events

var ignoredResources = map[schema.GroupResource]struct{}{
	{Group: "", Resource: "events"}: {},
}

ignoredResources group miss match with api-resource

events.k8s.io/v1

### What did you expect to happen?

ignore sync event

### How can we reproduce it (as minimally and precisely as possible)?

etcd have event key more than 4 million
restart kube-apiserver or controller-manager

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
v1.20.7
```

</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by niuyueyang1996**
events group fix

#### Which issue(s) this PR fixes:

Fixes #108318

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/controller/garbagecollector/graph_builder.go [hunk 1]
```diff
@@ -317,8 +317,11 @@ func (gb *GraphBuilder) Run(stopCh <-chan struct{}) {
 	klog.Infof("stopped %d of %d monitors", stopped, len(monitors))
 }
 
+// Event Group v1 events.k8s.io/v1
 var ignoredResources = map[schema.GroupResource]struct{}{
-	{Group: "", Resource: "events"}: {},
+	{Group: "", Resource: "events"}:                 {},
+	{Group: "v1", Resource: "events"}:               {},
```
Comment (tossmilestone) [inline pkg/controller/garbagecollector/graph_builder.go:323]: `v1` is the `version` but not `group`, the line should be removed.

File: pkg/controller/garbagecollector/graph_builder.go [hunk 2]
```diff
@@ -317,8 +317,11 @@ func (gb *GraphBuilder) Run(stopCh <-chan struct{}) {
...
+	{Group: "events.k8s.io/v1", Resource: "events"}: {},
```
Comment (tossmilestone) [inline pkg/controller/garbagecollector/graph_builder.go:324]: The group of `events.k8s.io/v1` should be `events.k8s.io`.

**Review by tossmilestone — CHANGES_REQUESTED** [review]

**Review by tossmilestone — APPROVED** [review]

**Comment by ialidzhikov** [comment]
release-v1.20 branch is no longer active. I doubt that there will be a new patch for v1.20. You can check the master branch and check whether the fix is applicable there first.

**Comment by niuyueyang1996** [comment]
#108404

**Comment by pacoxu** [comment]
I think you should merge it to the master branch at first and then do cherry-pick for 1.21-1.23 if needed.
For v1.20, the end of life on the website is yesterday and you may check with @kubernetes/release-managers.

**Comment by niuyueyang1996** [comment]
I created a new pr merge it to master.
#108404

**Comment by pacoxu** [comment]
/retitle ignoredResources should contain events.k8s.io events
/release-note-none

**Comment by justaugustus** [comment]
(Closing as v1.20 is [no longer a supported Kubernetes release series](https://kubernetes.io/releases/patch-releases/#non-active-branch-history).)

**Pull request closed without merging**